### PR TITLE
fix the unit test of clock

### PR DESCRIPTION
**Pull request by JustXxx**
forge test --mc FaultDisputeGameN_Test --mt .*clock.*

### CODE REVIEW

**Review by qizhou — APPROVED** [review]

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 1]
```diff
@@ -486,34 +486,15 @@ contract FaultDisputeGameN_Test is FaultDisputeGame_Init {
         uint64 halfGameDuration = gameProxy.maxClockDuration().raw();
         uint64 clockExtension = gameProxy.clockExtension().raw();
 
-        // Make an initial attack against the root claim with 1 second left on the clock. The grandchild should be
-        // allocated exactly `clockExtension` seconds remaining on their potential clock.
+        // The move above the split depth's grand child is the execution trace bisection root. The grandchild should
```
Comment (pochenai) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: "The move above the split depth's grand child" means the move is located at `split depth's grand child` (`rootclaim`), but your meaning seems to be the move's grand child is the `rootclaim`. To be more clear, I would suggest adding comment like this "This specific move, located above the split depth, has a grandchild that acts as the multi-section root in the execution trace. Consequently, if the current move has less than `clockExtension` seconds remaining on its potential clock, the grandchild move should be allocated `clockExtension` * 2 seconds."

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 2]
```diff
@@ -430,10 +430,10 @@ contract FaultDisputeGameN_Test is FaultDisputeGame_Init {
     function test_move_clockTimeExceeded_reverts() public {
         // Warp ahead past the clock time for the first move (3 1/2 days)
         vm.warp(block.timestamp + 3 days + 12 hours + 1);
-        uint256 bond = _getRequiredBond(0);
+        uint256 bond = _getRequiredBondV2(0, 0);
```
Comment (dajuguan) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol:433]: Should we slash the full claim or one branch of the claim?

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 3]
```diff
@@ -462,22 +462,22 @@ contract FaultDisputeGameN_Test is FaultDisputeGame_Init {
...
+        bond = _getRequiredBondV2(2, 0);
         (,,,, disputed,,) = gameProxy.claimData(2);
-        gameProxy.attack{ value: bond }(disputed, 2, claim);
+        gameProxy.attackV2{ value: bond }(disputed, 2, claim, 0);
         (,,,,,, clock) = gameProxy.claimData(3);
         assertEq(clock.raw(), LibClock.wrap(Duration.wrap(25), Timestamp.wrap(uint64(block.timestamp))).raw());
 
         vm.warp(block.timestamp + 10);
-        bond = _getRequiredBond(3);
+        bond = _getRequiredBondV2(3, 0);
```
Comment (pochenai) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol:482]: As in OP's [bond spec](https://specs.optimism.io/fault-proof/stage-one/bond-incentives.html):
> FPM bonds are priced in the amount of gas that they are intended to cover. Bonds start at the very first depth of the game at a baseline of 400_000 gas. The 400_000 value is chosen as a deterrence amount that is approximately double the cost to respond at the top level.

Due to more claims (~2048 claims) should be submitted in multi-sec fault proof, maybe we should double check the gas at the top level remains ~200_000 gas cost later?

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 4]
```diff
@@ -2461,6 +2442,27 @@ contract FaultDisputeGameN_LessSplitDepth_Test is FaultDisputeGame_Init {
...
         gameProxy.stepV2({ _claimIndex: 4, _attackBranch: 3, _stateData: claimData6, _proof: stepProof });
     }
+
+    /// @notice Static unit test that checks proper clock extension.
+    function test_move_clockExtensionCorrectnessOne_succeeds() public {
+        (,,,,,, Clock clock) = gameProxy.claimData(0);
+        assertEq(clock.raw(), LibClock.wrap(Duration.wrap(0), Timestamp.wrap(uint64(block.timestamp))).raw());
+
+        Claim claim = _dummyClaim();
+        uint256 splitDepth = gameProxy.splitDepth();
```
Comment (pochenai) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Maybe remove unused `splitDepth`?

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 5]
```diff
@@ -462,22 +462,22 @@ contract FaultDisputeGameN_Test is FaultDisputeGame_Init {
...
         (,,,, disputed,,) = gameProxy.claimData(3);
-        gameProxy.attack{ value: bond }(disputed, 3, claim);
+        gameProxy.attackV2{ value: bond }(disputed, 3, claim, 0);
         (,,,,,, clock) = gameProxy.claimData(4);
         assertEq(clock.raw(), LibClock.wrap(Duration.wrap(20), Timestamp.wrap(uint64(block.timestamp))).raw());
     }
 
     /// @notice Static unit test that checks proper clock extension.
-    function test_move_clockExtensionCorrectness_succeeds() public {
+    function test_move_clockExtensionCorrectnessSplitGrandChild_succeeds() public {
```
Comment (pochenai) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol:490]: Maybe remove unused `splitDepth`?

Comment (JustXxx) [reply packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol:490]: removed

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol (hunk 4 above)
Comment (JustXxx) [reply packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: removed

**Pull request merged**